packages/builders/src/components/selectMenu/ChannelSelectMenu.ts

```diff
@@ -59,6 +64,39 @@ export class ChannelSelectMenuBuilder extends BaseSelectMenuBuilder<APIChannelSe
 		return this;
 	}
 
+	/**
+	 * Adds default channels to this auto populated select menu.
+	 *
+	 * @param channels - The channels to add
+	 */
+	public addDefaultChannels(...channels: RestOrArray<Snowflake>) {
+		const normalizedValues = normalizeArray(channels).map((id) => ({
+			id,
+			type: SelectMenuDefaultValueType.Channel as const,
+		}));
+
+		this.data.default_values ??= [];
+		optionsLengthValidator.parse(this.data.default_values.length + normalizedValues.length);
+		this.data.default_values.push(...normalizedValues);
```

Comment:
Changing the operation order to not map or ensure `default_values` will reduce overhead in the case of the validator throwing an error.

- normalizeArray → validate length → ensure array → map values → push to array

Reply:
Should be done with fa4833ceec160487b6ea423548fdac317874f7f1.